File: s1/L3/python/prime.py
import sys
import packages.primenumbers as pp

def main():
    if len(sys.argv) != 2:
        print("Zła liczba argumentów")
        return
    n = int(sys.argv[1])
    if n < 0:
        print("Liczby nie mogą być mniejsze od 0")
        return
    if n < 2:
        print("0")
        return
    s = []
    pp.create_sieve(s, n)
    pp.compute_sieve(s)
    c = pp.count_primes(s)
    print(c)

if __name__ == "__main__":
    main()

File: s1/L3/c/prime.c
#include "prime.h"
#include <stddef.h>

int compute_sieve(bool s[], unsigned n){
    if(s == NULL) return 1;
    if(n < 2) return 1;

    unsigned i, j;
    for (i = 2; i <= n; i++) s[i] = true;
    for (i = 2; i <= n; i++)
        if (s[i])
            for (j = i + i; j <= n; j += i)
                s[j] = false;
    return 0;
}

unsigned count_primes(bool s[], unsigned n){
    int i, c = 0;
    for (i = 2; i <= n; i++)
        if(s[i]) c++;
    return c;
}
